libcxx/include/__format/format_parse_context.h

```diff
@@ -83,13 +84,16 @@ class _LIBCPP_TEMPLATE_VIS basic_format_parse_context {
       std::__throw_format_error("Argument index outside the valid range");
   }
 
+  _LIBCPP_HIDE_FROM_ABI constexpr bool& __should_parse() { return __parse_; }
+
 private:
   iterator __begin_;
   iterator __end_;
   enum _Indexing { __unknown, __manual, __automatic };
   _Indexing __indexing_;
   size_t __next_arg_id_;
   size_t __num_args_;
+  bool __parse_;
```

Comment:
FYI this (likely) changes the size of a public type which results in an ABI break. This is something that we should try very hard to avoid. A possible solution would be to use an unused bit of `__indexing_` or other padding bits. With the proposed approach we don't need this boolean, I just wanted to mention this generic Standard library rule.

Reply:
Thanks for the heads up. I was thinking very hard about this and trying to squeeze this information into the most significant bit of `__num_args_`. But I have since realized this is barking up the wrong tree.